statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,245 @@
+                    cluster_stats[ind] -= self._total[column_index]
+                elif method == "mean":
+                    cluster_stats[ind] = np.array(np.dot(self.data[:, column_index], new_weights).item())
+                    cluster_stats[ind] /= np.sum(new_weights)
+                    cluster_stats[ind] -= self._mean[column_index]
+
+            stratum_stats[id] = np.sum((cluster_stats) ** 2)
+            stratum_stats[id] *= (num_clusters - 1) / num_clusters
+        return np.sqrt(np.sum(stratum_stats))
+
+            # diff = (cluster_stats - method) * (num_clusters) / (num_clusters - 1)
+            # stratum_stats[id] = sum(diff)
+        
+
+    def _reweight(self, cluster, id_stratum, method):
+        # make sure to throw and error if len(num_clusters == 1) 
+        # ie you can't calc a statistic minus a cluster bc there's only one
+        num_clusters = len(np.unique(self.cluster[id_stratum]))
+        # in stratum h but not in cluster j
+        # print("cluster #:", cluster)
+        # print("cluster indices:", np.where(self.cluster == cluster)[0])
+        # print("stratum indices:", id_stratum)
+        id_noncluster = np.intersect1d(id_stratum, np.where(self.cluster != cluster)[0])
+
+        # in stratum h and in cluster j
+        id_cluster = np.intersect1d(id_stratum, np.where(self.cluster == cluster)[0])
+
+        new_weights = np.array(self.prob_weights, copy=True)
+        new_weights[id_noncluster] = (num_clusters / (num_clusters - 1)) * self.prob_weights[id_noncluster]
+        new_weights[id_cluster] = 0
+
+        return new_weights        
+
+
+    def show(self):
+        print(self.data)
+        print(self.cluster)
+        print(self.strata)
+        print(self.prob_weights)
+
+    def total(self, method):
+        """
+        Calculates the total for each variable. 
+
+        Parameters
+        ----------
+        self
+
+        Returns
+        -------
+        A tuple of length 2, each index is a 1xp array. The first array holds 
+        the total for each of the p vars in self.data. The second array is the
+        SE for each total
+        """
+
+        # creates "dummy" arrays for prob_weights, cluster, and strata
+        # if any or all were not supplied
+        n = self.data.shape[0]
+        if not isinstance(self.prob_weights, np.ndarray):
+            self.prob_weights = np.ones([n,1])
+        if not isinstance(self.cluster, np.ndarray):
+            self.cluster = np.ones([n,1])
+        if not isinstance(self.strata, np.ndarray):
+            self.strata = np.ones([n,1])
+
+        self._total = [np.dot(self.data[:, index], self.prob_weights).item() for index in range(self.data.shape[1])]
+        self._total = np.array(self._total)
+
+        if method == "jack":
+          total_se = np.array([self._jackknife("total", index) for index in range(self.data.shape[1])])
+        
+        return self._total, total_se
+
+
+    def mean(self, method):
```

Comment:
Ah, I see your design here. Mine differs only in that I call _stat in a list comprehension, instead of calling _stat once and using a list comprehension in it